themes/classic/_dev/js/cart.js

```diff
@@ -168,6 +168,36 @@ $(document).ready(() => {
     });
   };
 
+  //Hide 'have a code promo' text if displayed when 'close' is
+  let isPresent = $('.promo-code-button').length,
+      nameLength = 0;
+
+  function labelCodeDisplayed() {
+    $('.promo-code-button').not('.cancel-promo').hide();
+    
+    $('.cancel-promo').on('click', () => {
+      $('.cancel-promo').hide();
+      $('.promo-code-button').not('.cancel-promo').show();
+    });
+    $('.promo-code-button').not('.cancel-promo').on('click', () => {
+        $('.promo-code-button').not('.cancel-promo').hide();
+        $('.cancel-promo').show();
+    });
+  }
+
+  setInterval( function () {
+    if ($('.promo-name').length != nameLength){
+      nameLength = $('.promo-name').length;
```

Comment:
`.length` is not chached by the javascript engine. It would be great to store this value in a variable